### PR TITLE
1. Split exploration and patrolling into different prefabs

### PR DESCRIPTION
**Pull request by Puvikaran2001**
This PR splits exploration and patrolling into different prefabs, and thereby avoids set button and other manually through code, but can be setup easily in Unity.

Also avoid destroying manually

There is some bug with the patrolling visualizer, that would be fix in another PR :)

### CODE REVIEW

File: Assets/Scripts/Simulation/ExplorationSimulationManager.cs [hunk 1]
```diff
@@ -9,14 +9,6 @@ namespace MAES.Simulation
...
-        public override void AddRestartRemakeController(GameObject restartRemakePanel)
-        {
-            restartRemakePanel.AddComponent<ExplorationRestartRemakeController>();
-        }
-
-        protected override ExplorationSimulation AddSimulation(GameObject gameObject)
-        {
-            return gameObject.AddComponent<ExplorationSimulation>();
-        }
+
```
Comment (henneboy) [inline Assets/Scripts/Simulation/ExplorationSimulationManager.cs:12]:
```suggestion
```

**Review by henneboy — APPROVED** [review]

**Review by miniJmo — APPROVED** [review]
:shipit:

File: Assets/Scripts/UI/SimulationInfoUIControllers/ExplorationInfoUIController.cs [hunk 2]
```diff
@@ -30,19 +30,6 @@ public sealed class ExplorationInfoUIController : SimulationInfoUIControllerBase
 
         protected override void AfterStart()
         {
-            ExplorationBarMask = GameObject.Find("ExplorationBarMask").GetComponent<Image>();
```
Comment (Casper-NS) [inline Assets/Scripts/UI/SimulationInfoUIControllers/ExplorationInfoUIController.cs:33]: Arent these needed to make the UI work?

**Review by Casper-NS — COMMENTED** [review]
Looks fine though i would like to have a quick look at it together to discuss some things

**Review by Casper-NS — APPROVED** [review]
lgtm

**Pull request merged**